lib/src/model/broadcast/broadcast_repository.dart

```diff
@@ -51,34 +71,75 @@ BroadcastsList _makeBroadcastResponseFromJson(
 }
 
 Broadcast _broadcastFromPick(RequiredPick pick) {
-  final live = pick('round', 'ongoing').asBoolOrFalse();
-  final finished = pick('round', 'finished').asBoolOrFalse();
-  final status = live
-      ? RoundStatus.live
-      : finished
-          ? RoundStatus.finished
-          : RoundStatus.upcoming;
   final roundId = pick('round', 'id').asBroadcastRoundIdOrThrow();
 
   return Broadcast(
-    tour: (
-      name: pick('tour', 'name').asStringOrThrow(),
-      imageUrl: pick('tour', 'image').asStringOrNull(),
-    ),
-    round: BroadcastRound(
-      id: roundId,
-      name: pick('round', 'name').asStringOrThrow(),
-      status: status,
-      startsAt: pick('round', 'startsAt')
-          .asDateTimeFromMillisecondsOrThrow()
-          .toLocal(),
-    ),
+    tour: _tournamentDataFromPick(pick('tour').required()),
+    round: _roundFromPick(pick('round').required()),
     group: pick('group').asStringOrNull(),
     roundToLinkId:
         pick('roundToLink', 'id').asBroadcastRoundIddOrNull() ?? roundId,
   );
 }
 
+BroadcastTournamentData _tournamentDataFromPick(
+  RequiredPick pick,
+) =>
+    BroadcastTournamentData(
+      id: pick('id').asBroadcastTournamentIdOrThrow(),
+      name: pick('name').asStringOrThrow(),
+      imageUrl: pick('image').asStringOrNull(),
+      description: pick('description').asStringOrNull(),
+      information: (
+        format: pick('info', 'format').asStringOrNull(),
+        timeControl: pick('info', 'tc').asStringOrNull(),
+        players: pick('info', 'players').asStringOrNull(),
+        dates: pick('dates').letOrNull(
+          (pick) => (
+            startsAt: pick(0).asDateTimeFromMillisecondsOrThrow().toLocal(),
```

Comment:
I realise now the
```dart
      return DateTime.fromMillisecondsSinceEpoch(value, isUtc: true);
```

from `asDateTimeFromMillisecondsOrThrow` is a mistake; We should always return local time, not UTC. So I'd rather remove the `isUtc: true` to have all the date handled the same (and remove the `toLocal()` here).

Reply:
The server returns UTC so we need to keep `isUtc: true`, right ?

However I can move the `toLocal()` in the pick extension

Reply:
The server returns a timestamp representing the number of milliseconds since epoch, which is always using UTC as a reference, that is true.

But the [dart constructor](https://api.dart.dev/stable/3.5.3/dart-core/DateTime/DateTime.fromMillisecondsSinceEpoch.html) `isUtc: true` parameter is a configuration to decide whether we want a `DateTime` in local time or in UTC. We always want to display the time in local time that's why the `isUtc` has to be false, which is the default.

Reply:
Oh right, I'll remove the `isUtc: true`